### PR TITLE
Make config overrides work with Mojolicious::Lite

### PR DESCRIPTION
**Pull request by kraih**
This is a solution for #1406.

### CODE REVIEW

File: lib/Mojolicious.pm [hunk 1]
```diff
@@ -57,6 +57,8 @@ has validator => sub { Mojolicious::Validator->new };
 our $CODENAME = 'Waffle';
 our $VERSION  = '9.07';
 
+our @ARGS_OVERRIDE;
```
Comment (kraih) [inline lib/Mojolicious.pm:60]: This is pretty ugly, and i would welcome suggestions for a cleaner solution.

**Review by christopherraa — COMMENTED** [review]
The overall change does look good I think. It seems like a pretty minimal change and solves the problem at hand in a good way. Consider this an approval, but I am holding off on actually approving until the thread asking for a better solution has been resolved (so as to avoid auto-merging prematurely). If you feel that the solution is OK as it is then I'll add my approval.

**Comment by kraih** [comment]
To prevent auto-merging a PR can also just be labeled work in progress.

**Comment by kraih** [comment]
My main issue was bad encapsulation with the global variable, `Mojo::Server` had to require `Mojolicious` to be able to set `@Mojolicious::ARGS_OVERRIDE`. Now i've reversed that and the global variable is `@Mojo::Server::ARGS_OVERRIDE`. That is at least a little better.

**Review by christopherraa — APPROVED** [review]
I am now :+1: to this. Agree that it much better with the args-move.

**Review by marcusramberg — APPROVED** [review]
Looks pretty clean now, and useful feature

**Pull request merged**